Repository/ServiceEntityRepository.php

```diff
@@ -3,57 +3,38 @@
 namespace Doctrine\Bundle\DoctrineBundle\Repository;
 
 use Doctrine\ORM\EntityRepository;
-use Doctrine\Persistence\ManagerRegistry;
-use LogicException;
 use Symfony\Component\VarExporter\LazyGhostTrait;
 
-use function sprintf;
+use function property_exists;
 use function trait_exists;
 
-if (trait_exists(LazyGhostTrait::class)) {
-    /**
-     * @template T of object
-     * @template-extends LazyServiceEntityRepository<T>
-     */
-    class ServiceEntityRepository extends LazyServiceEntityRepository
-    {
+if (property_exists(EntityRepository::class, '_entityName')) {
```

Comment:
Oh boy, people were already complaining about previous logic being too difficult for SA tools to understand (https://github.com/doctrine/DoctrineBundle/issues/1677), now it's going to get worse. But I guess we can't do anything about it

Reply:
Not without breaking stuff, I'm afraid. It might be worth doing a 3.0 soon'ish.